src/main/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/sideeffects/CardEventHandler.java

```diff
@@ -13,39 +10,73 @@
 public class CardEventHandler {
   private final CardRepository cardRepo;
   private final RoundRepository roundRepo;
+  private final MatchRepository matchRepo;
+  private final HandRepository handRepo;
+  private final GameRepository gameRepo;
+  private InstanceCreator instanceCreator;
 
-  public CardEventHandler(RoundRepository roundRepo, CardRepository cardRepo) {
+  public CardEventHandler(
+      RoundRepository roundRepo,
+      CardRepository cardRepo,
+      MatchRepository matchRepo,
+      HandRepository handRepo,
+      GameRepository gameRepo) {
     this.roundRepo = roundRepo;
     this.cardRepo = cardRepo;
+    this.matchRepo = matchRepo;
+    this.handRepo = handRepo;
+    this.gameRepo = gameRepo;
   }
 
   @SuppressWarnings("unused")
   @HandleAfterSave
   @Transactional
   public void handleAfterSave(Card card) {
     Round currentRound = card.getRound();
```

Comment:
```suggestion
  Round round = card.getRound();
```
Ob die round current ist, weist du zu diesem Zeitpunkt nicht.
Es könnte auch sein, dass jemand den kartenwert patcht... oder etwas anderes.